src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TableMarshallerProvider.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx
+{ 
+    /// <summary>
+    /// Marshal .Net classes (with fields). This supports strong typing and lazy marshalling. 
+    /// Handles any IEnumerable (including arrays).
+    /// </summary>
+    public class TableMarshallerProvider : ITypeMarshallerProvider
+    {
+        /// <inheritdoc/>
+        public bool TryGetMarshaller(Type type, TypeMarshallerCache cache, int maxDepth, out ITypeMarshaller marshaler)
+        {
+            if (!IsIEnumerableT(type, out var et))
+            {
+                marshaler = null;
+                return false;
+            }
+
+            var rowMarshaller = cache.GetMarshaller(et, maxDepth);
+             
+            if (rowMarshaller.Type is RecordType recordType)
+            {
+                // Array of records 
+            }       
+            else
```

Comment:
consider structuring this as `if (rowMarshaller.Type is not RecordType recordType)` and avoid the empty if block